packages/server/src/fhir/operations/codesystemvalidatecode.ts

```diff
@@ -62,19 +62,38 @@ export async function codeSystemValidateCodeHandler(req: FhirRequest): Promise<F
 }
 
 export async function validateCoding(codeSystem: CodeSystem, coding: Coding): Promise<Coding | undefined> {
-  if (coding.system && coding.system !== codeSystem.url) {
-    return undefined;
+  return validateCodings(codeSystem, [coding]).then((results) => results[0]);
+}
+
+export async function validateCodings(codeSystem: CodeSystem, codings: Coding[]): Promise<(Coding | undefined)[]> {
+  const eligible: boolean[] = new Array(codings.length);
+  const codesToQuery = new Set<string>();
+  for (let i = 0; i < codings.length; i++) {
+    const c = codings[i];
+    if (c.system && c.system !== codeSystem.url) {
+      continue;
+    }
+    if (c.code) {
+      codesToQuery.add(c.code);
+      eligible[i] = true;
+    }
   }
 
-  const query = new SelectQuery('Coding')
-    .column('id')
-    .column('display')
-    .where('code', '=', coding.code)
-    .where('system', '=', codeSystem.id);
+  let result: any[] | undefined;
```

Comment:
You could use `Coding` rather than `any` here, because it has `id`, `code`, and `display` as optional string fields.